src/main/java/org/assertj/core/internal/Iterables.java

```diff
@@ -1137,10 +1137,7 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
       // if the objects are not equal, begin the error handling process
       if (!areEqual(actualAsList.get(i), values[i])) {
         IterableDiff<Object> diff = diff(actualAsList, asList(values), comparisonStrategy);
-        if (diff.differencesFound()) {
-          throw shouldContainExactlyWithDiffAssertionError(diff, actual, values, info);
-        }
-        throw failures.failure(info, elementsDifferAtIndex(actualAsList.get(i), values[i], i, comparisonStrategy));
+        throw shouldContainExactlyWithDiffAssertionError(diff, actual, values, info);
```

Comment:
Does it make sense to include the first index of the diff elements into the message as well?